sdk/internal/recording/recording.go

```diff
@@ -486,20 +482,22 @@ func (r RecordingOptions) HostScheme() string {
 }
 
 func getTestId(pathToRecordings string, t *testing.T) string {
+	if strings.HasSuffix(pathToRecordings, "/") {
+		pathToRecordings = strings.TrimRight(pathToRecordings, "/")
+	}
 	return pathToRecordings + "/recordings/" + t.Name() + ".json"
 }
 
 func StartRecording(t *testing.T, pathToRecordings string, options *RecordingOptions) error {
 	if options == nil {
 		options = defaultOptions()
 	}
-	if recordMode == "" {
-		t.Log("AZURE_RECORD_MODE was not set, options are \"record\" or \"playback\". \nDefaulting to playback")
-		recordMode = "playback"
+	if !(recordMode == modeRecording || recordMode == modePlayback) {
+		return fmt.Errorf("AZURE_RECORD_MODE was not understood, options are \"record\" or \"playback\". Received: %v", recordMode)
```

Comment:
No biggie, but could '%s' and use the actual constants in your error message since you're already Errorf'ing.